awx/ui_next/src/components/Schedule/ScheduleAdd/ScheduleAdd.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { func } from 'prop-types';
+import { t } from '@lingui/macro';
+import { withI18n } from '@lingui/react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { RRule } from 'rrule';
+import { Card } from '@patternfly/react-core';
+import { CardBody } from '@components/Card';
+import { getWeekNumber } from '@util/dates';
+import ScheduleForm from '../shared/ScheduleForm';
+
+const days = {
+  0: 'SU',
+  1: 'MO',
+  2: 'TU',
+  3: 'WE',
+  4: 'TH',
+  5: 'FR',
+  6: 'SA',
+};
+
+function ScheduleAdd({ i18n, createSchedule }) {
+  const [formSubmitError, setFormSubmitError] = useState(null);
+  const history = useHistory();
+  const location = useLocation();
+  const { pathname } = location;
+  const pathRoot = pathname.substr(0, pathname.indexOf('schedules'));
+
+  const handleSubmit = async values => {
+    try {
+      const [startDate, startTime] = values.startDateTime.split('T');
+      // Dates are formatted like "YYYY-MM-DD"
+      const [startYear, startMonth, startDay] = startDate.split('-');
+      // Times are formatted like "HH:MM:SS" or "HH:MM" if no seconds
+      // have been specified
+      const [startHour = 0, startMinute = 0, startSecond = 0] = startTime.split(
+        ':'
+      );
+
+      const ruleObj = {
```

Comment:
please pull this into a function
```js
function createRuleObj(values) {
  ...
  return ruleObj;
```

Reply:
https://github.com/ansible/awx/pull/6419/commits/d72a5d85a17fc84ade0b6459069cfad14e9f8eb4